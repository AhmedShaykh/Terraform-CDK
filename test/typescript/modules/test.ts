/**
 * Testing interaction with Terraform Cloud
 *
 * @group typescript
 */

import { TestDriver } from "../../test-helper";
import * as fs from 'fs-extra';
import * as path from 'path';

const onWindows = process.platform === 'win32' ? it.skip : it.skip
const onPosix = process.platform !== 'win32' ? it : it.skip

describe("full integration test", () => {
  let driver: TestDriver;

  beforeAll(() => {
    driver = new TestDriver(__dirname)
    driver.switchToTempDir()
    driver.init('typescript')
    driver.copyFiles('main.ts', 'cdktf.json')
    fs.copySync(path.join(__dirname, 'local-module'), path.join(driver.workingDirectory, 'local-module'))
    driver.get()
  });

  onPosix("build modules posix", () => {
    driver.synth()
    expect(driver.synthesizedStack()).toMatchSnapshot()
  })

  onWindows("build modules windows", () => {
    driver.synth()
    expect(driver.synthesizedStack()).toMatchSnapshot()
  })
})