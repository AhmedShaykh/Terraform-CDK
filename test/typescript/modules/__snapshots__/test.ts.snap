// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full integration test build modules posix 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"hello-modules\\",
      \\"backend\\": \\"local\\"
    }
  },
  \\"module\\": {
    \\"localmodule\\": {
      \\"source\\": \\"./assets/localmodulelocalmodule/97F51B069A2EA5F1B6BA8B6C0FED84E4\\",
      \\"//\\": {
        \\"metadata\\": {
          \\"path\\": \\"hello-modules/local-module\\",
          \\"uniqueId\\": \\"localmodule\\"
        }
      }
    },
    \\"gcloud\\": {
      \\"source\\": \\"terraform-google-modules/gcloud/google\\",
      \\"version\\": \\"2.0.3\\",
      \\"//\\": {
        \\"metadata\\": {
          \\"path\\": \\"hello-modules/gcloud\\",
          \\"uniqueId\\": \\"gcloud\\"
        }
      }
    },
    \\"iam\\": {
      \\"account_alias\\": \\"cdktf\\",
      \\"source\\": \\"terraform-aws-modules/iam/aws//modules/iam-account\\",
      \\"version\\": \\"3.12.0\\",
      \\"//\\": {
        \\"metadata\\": {
          \\"path\\": \\"hello-modules/iam\\",
          \\"uniqueId\\": \\"iam\\"
        }
      }
    }
  }
}"
`;

exports[`full integration test build modules windows 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"hello-modules\\",
      \\"backend\\": \\"local\\"
    }
  },
  \\"module\\": {
    \\"localmodule\\": {
      \\"source\\": \\"..\\\\\\\\..\\\\\\\\..\\\\\\\\local-module\\",
      \\"//\\": {
        \\"metadata\\": {
          \\"path\\": \\"hello-modules/local-module\\",
          \\"uniqueId\\": \\"localmodule\\"
        }
      }
    },
    \\"gcloud\\": {
      \\"source\\": \\"terraform-google-modules/gcloud/google\\",
      \\"version\\": \\"2.0.3\\",
      \\"//\\": {
        \\"metadata\\": {
          \\"path\\": \\"hello-modules/gcloud\\",
          \\"uniqueId\\": \\"gcloud\\"
        }
      }
    },
    \\"iam\\": {
      \\"account_alias\\": \\"cdktf\\",
      \\"source\\": \\"terraform-aws-modules/iam/aws//modules/iam-account\\",
      \\"version\\": \\"3.12.0\\",
      \\"//\\": {
        \\"metadata\\": {
          \\"path\\": \\"hello-modules/iam\\",
          \\"uniqueId\\": \\"iam\\"
        }
      }
    }
  }
}"
`;
